pythran/pythonic/numpy/array_equal.hpp

```diff
@@ -9,24 +9,12 @@ namespace pythonic {
 
     namespace numpy {
         template<class U, class V>
-            typename std::enable_if<types::has_shape<U>::value and types::has_shape<V>::value,bool>::type array_equal(U const& u, V const&v) {
+            bool array_equal(U const& u, V const&v) {
                 if(u.shape == v.shape) {
-                    return all(u == v);
+                    return all(types::numpy_expr<operator_::proxy::eq, U, V>(u, v));
```

Comment:
Because operator== will return a boolean for list which is not iterable while this numpy_expr notation is the same for all arraylike.